frontend/src/timeline.tsx

```diff
@@ -0,0 +1,151 @@
+import classes from './timeline.module.css';
+import Event from './Event/Event';
+import {
+  DayInfo, 
+  TimelineInfo
+} from '../../backend/src/interfaces';
+
+const dayMonthOption = { day: 'numeric', month: 'numeric' };
+const weekdayOption = { weekday: 'long' };
+
+const getDate = function (date: Date, options: { }): string {
+  return date.toLocaleDateString('en-GB', options);
+};
+
+const startDate = new Date(2024, 9, 28);
+const endDate = new Date(2024, 10, 3);
+const days = Array.from({ length: 7 }, (e, i) => {
+
+  e = 50; //straight up this line is just here to get rid of "value declared but not read" warning. 
+  // it was bugging me
+
+  const date: Date = new Date(startDate.valueOf());
+
+  date.setDate(startDate.getDate() + i);
+
+  return {
+    date,
+    events: []
+  };
+});
+
+function Day(dayProps: DayInfo) {
+
+  return (
+    <div className={classes.container}>
+      {
+        dayProps.events.map((event) => Event(event))
+      }
+    </div>
+  );
+}
+
+function TimelineTitle(TimelineTitleProps: TimelineInfo) {
+  const { startDate, endDate } = TimelineTitleProps;
+
+  return (
+    <div className={classes.timelineTitle}>
+      <p className={classes.weekTitle}>{'Week 8' /*I have no clue what to put here */}</p>
```

Comment:
'week 8' is fine for now. note that you don't need to use {} and ''; you can just write Week 8 like any other HTML element